src/server/qtquick/wsurfaceitem.cpp

```diff
@@ -366,11 +368,33 @@ void WSurfaceItemContent::componentComplete()
         d->init();
 }
 
+class WSGRenderFootprintNode: public QSGRenderNode
+{
+public:
+    WSGRenderFootprintNode(WSurfaceItemContent *owner)
+        : QSGRenderNode()
+        , m_owner(owner)
+    {
+        setFlag(QSGNode::OwnedByParent); // parent is fixed, auto release
+    }
+
+    ~WSGRenderFootprintNode() {}
+
+    void render(const RenderState*) override
+    {
+        m_owner->rendered = true;
+    }
+
+    WSurfaceItemContent *m_owner;
+};
+
 QSGNode *WSurfaceItemContent::updatePaintNode(QSGNode *oldNode, UpdatePaintNodeData *)
 {
+    auto privt=QQuickItemPrivate::get(this);
     W_D(WSurfaceItemContent);
     if (!d->textureProvider || !d->textureProvider->texture() || width() <= 0 || height() <= 0) {
-        delete oldNode;
+        if (oldNode)
```

Comment:
Not necessarily

Reply:
add assertion? below there's a QUNLIKELY(!node)

Reply:
No, `delete nullptr` is uninjurious

Reply:
🤯